corehq/ex-submodules/casexml/apps/phone/tests/test_ota_restore.py

```diff
@@ -131,6 +131,11 @@ def testOverwriteCache(self):
         self.assertIsInstance(restore_config_cached.get_payload(), CachedResponse)
         self.assertNotIsInstance(restore_config_overwrite.get_payload(), CachedResponse)
 
+        # even cached responses change the sync log id so they are not the same
```

Comment:
here is the crux of the `finalize` problem in the `CachedPayload`. this will continue to fail because we don't replace the sync log id in the restore. i really want to avoid having to do that. i think we discussed that we can just add the device_id to the restore_cache_key and that will alleviate the issue (have not done that yet). it's a bit confusing to me still cause i'm not exactly sure why it's bad when a user is logged into multiple devices and gets the same restore payload